envoy/common/scope_tracker.h

```diff
@@ -18,6 +19,8 @@ class ScopeTrackedObject {
 public:
   virtual ~ScopeTrackedObject() = default;
 
+  virtual ScopedExecutionContext scoped_execution_context() const PURE;
+
```

Comment:
Seems most `ScopeTrackedObject`s only return an empty `ScopedExecutionContext`, may be you can provide a default implementation here to avoid spread the `ScopedExecutionContext` everywhere.

I believe only limited users may need this new feature.